src/components/LoadingIcon/PLoadingIcon.vue

```diff
@@ -1,19 +1,38 @@
 <template>
-  <div class="p-loading-icon">
+  <div class="p-loading-icon" :class="iconSizeClass">
     <div class="p-loading-icon__segment" />
     <div class="p-loading-icon__segment" />
     <div class="p-loading-icon__segment" />
     <div class="p-loading-icon__segment" />
   </div>
 </template>
 
+<script setup lang="ts">
+  import { computed, defineProps } from 'vue'
+
+  const props = defineProps<{
+    size?: 'small' | 'large',
+  }>()
+  const iconSizeClass = computed(() => props.size && `p-loading-icon--${props.size}`)
```

Comment:
Let's not use boolean logic so we don't end up trying to bind `false` as a class. Should be fine to just always add the class as long as there is a default. 
```suggestion
  const iconSizeClass = computed(() => `p-loading-icon--${props.size ?? 
  'large'}`)
```

Reply:
TY 🙏